From 8f9e9109ddbb3ec796ca876acad92d8bfeae4a19 Mon Sep 17 00:00:00 2001
From: Robert Marko <robert.marko@sartura.hr>
Date: Fri, 6 Nov 2020 13:05:07 +0100
Subject: [PATCH 2/2] arm64: dts: ipq8074: add MDIO support

IPQ807x uses the same MDIO controller as IPQ40xx that has a
upstream driver.
So lets enable the MDIO as prerequisite of any kind of networking
support.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>
---
 arch/arm64/boot/dts/qcom/ipq8074.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/arch/arm64/boot/dts/qcom/ipq8074.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq8074.dtsi
@@ -300,6 +300,16 @@
 			};
 		};
 
+		mdio: mdio@90000 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			compatible = "qcom,ipq4019-mdio";
+			reg = <0x90000 0x64>;
+			clocks = <&gcc GCC_MDIO_AHB_CLK>;
+			clock-names = "mdio";
+			status = "disabled";
+		};
+
 		prng: qrng@e1000 {
 			compatible = "qcom,prng-ee";
 			reg = <0xe3000 0x1000>;
